File: rb/features.rb
# Pony SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module PonyFeatures
  def self.make_feature(name)
    case name
    when "base"
      PonyBaseFeature.new
    when "test"
      PonyTestFeature.new
    else
      PonyBaseFeature.new
    end
  end
end

File: py/features.py
# Pony SDK feature factory

from feature.base_feature import PonyBaseFeature
from feature.test_feature import PonyTestFeature


def _make_feature(name):
    features = {
        "base": lambda: PonyBaseFeature(),
        "test": lambda: PonyTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
